### DISCUSSION TITLE
Support for own Electron version / instance

### DISCUSSION THREAD

**Discussion by ChrisKuBa** [cypress-io/cypress#29730] [Questions and Help]
Hi,

How can I configure Cypress to use my own electron version / installation?
Because the version of electron is bundled with cypress outdated and also we ship our own fixed version of electron, its necessary to test against a specific electron version.

I try to inject a new browser config, but cypress will not connect. The path to the electron.exe ist valid in my configuration, but no process will start / the process exits with 4294967295.

```javascript
import { defineConfig } from 'cypress';

import Browser = Cypress.Browser;

const addLocalElectonConfig = (browsersConfig: Browser[]): Browser[] => [
    ...browsersConfig,
    {
        name: 'electron-locale',
        channel: 'stable',
        family: 'chromium',
        displayName: 'Electron Local',
        version: '126.0.6478.36',
        path: 'C:\\...\\node_modules\\electron\\dist\\electron.exe',
        majorVersion: '126',
        isHeaded: true,
        isHeadless: false,
    },
];

export default defineConfig({
    component: {
        setupNodeEvents(on, config) {
            return {
                ...config,
                browsers: addLocalElectonConfig(config.browsers),
            };
        },
    },
    e2e: {
        setupNodeEvents(on, config) {
            return {
                ...config,
                browsers: addLocalElectonConfig(config.browsers),
            };
        },
    },
});
```
`
cypress:launcher:browsers launching browser { browser: { name: 'electron-locale', channel: 'stable', family: 'chromium', displayName: 'Electron Local', version: '126.0.6478.36', path: 'C:\\...\\node_modules\\electron\\dist\\electron.exe', majorVersion: '126', isHeaded: true, isHeadless: false }, url: 'about:blank' } +0ms

cypress:launcher:browsers spawning browser with opts { ...  } +2ms

cypress:network:connect received error on connect, retrying { iteration: 0, delay: 100, err: Error: connect ECONNREFUSED 127.0.0.1:53525     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) { errno: -4078, code: 'ECONNREFUSED', syscall: 'connect', address: '127.0.0.1', port: 53525 } } +3s

cypress:network:connect received error on connect, retrying { iteration: 1, delay: 100, err: Error: connect ECONNREFUSED 127.0.0.1:53525     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) { errno: -4078, code: 'ECONNREFUSED', syscall: 'connect', address: '127.0.0.1', port: 53525 } } +102ms

**cypress:launcher:browsers electron-locale exited: { code: 4294967295, signal: null } +118ms**

cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 53516 } +7s
`
Thx

**Comment by portyu9** [comment]
Hi @ChrisKuBa 

An arbitrary Electron executable is **not interchangeable with Cypress's bundled Electron browser**.

Cypress's `electron` browser is a Cypress-managed/bundled Electron runtime with Cypress-specific launch and automation integration. Pointing a custom Chromium-family browser entry at:

```text
node_modules/electron/dist/electron.exe
```

does not turn a standalone Electron executable into that Cypress runtime.

That explains why a definition such as:

```js
{
  name: 'electron-locale',
  family: 'chromium',
  path: '.../electron.exe',
}
```

can start a process but then fail when Cypress tries to establish the browser automation connection.

For an externally supplied browser, use a supported standalone Chromium browser instead, such as:

- Chrome
- Chrome for Testing
- Chromium
- Edge

and select its executable/path where needed.

`before:browser:launch` can modify launch options, preferences, extensions and command-line arguments, but it does not provide a supported mechanism for replacing Cypress's Electron runtime with another Electron version.

There is also an important current update: Cypress now marks its bundled Electron test browser as **deprecated** and recommends moving runs to Chrome or another installed browser.

So if the reason you need a newer Electron is a Chromium/browser fix, the practical solution is normally to run the Cypress suite against a corresponding recent Chrome/Chromium version rather than trying to swap Cypress's Electron dependency.

If instead you mean that the **application under test is itself an Electron desktop application**, that is a separate use case: Cypress is not a general Electron desktop-app automation framework, and supplying the application's Electron executable as a Cypress browser will not make that work.

Please mark this answer as accepted if it helped, thank you.
